File: test3.py
#함수 정의
def add(x,y):
    return x+y

#람다 함수
add2 = lambda x,y : x+y

result = add(10,20)
result2 = add2(10,20)

print(result)
print(result2)

arr = [1, 2, 3]

f2 = lambda x : 2 * x
f3 = lambda x : 3 * x

def multiply2(arr, f) :
    result10 = []
    for ele in arr :
      result10.append(f(ele))
    return result10

print(multiply2(arr, f2))
print(multiply2(arr, f3))
print(multiply2(arr, lambda x : 4 * x))

File: test3.js
//함수선언문
function add(x, y) {
  return x + y;
}

//함수표현식
const add2 = function (x, y) {
  return x + y;
};

//화살표함수
const add3 = (x, y) => x + y;

const result = add(10, 20);
const result2 = add2(10, 20);
const result3 = add3(10, 20);

console.log(result, result2, result3);

{
  const arr = [1, 2, 3];
  const result10 = [];

  const f2 = x => 2 * x;
  const f3 = x => 3 * x;

  function multiply2(arr, f) {
    for (let i = 0; i < arr.length; i++) {
      result10[i] = f(arr[i]);
    }
    return result10;
  }
  console.log(multiply2(arr, f2));
  console.log(multiply2(arr, f3));
  console.log(multiply2(arr, x => 4 * x));
}
